pkg/nmi/server/server.go

```diff
@@ -74,14 +77,27 @@ func (s *Server) Run() error {
 // NOT originating from HostIP destined to metadata endpoint are
 // routed to NMI endpoint
 func (s *Server) updateIPTableRules() {
-	log.Infof("node: %s %s", s.NodeName, s.HostIP)
-	for range time.Tick(time.Second * time.Duration(s.IPTableUpdateTimeIntervalInSeconds)) {
-		log.Infof("node(%s) hostip(%s) metadataaddress(%s:%s) nmiport(%s)", s.NodeName, s.HostIP, s.MetadataIP, s.MetadataPort, s.NMIPort)
-		if err := iptables.AddCustomChain(s.MetadataIP, s.MetadataPort, s.HostIP, s.NMIPort); err != nil {
-			log.Fatalf("%s", err)
-		}
-		if err := iptables.LogCustomChain(); err != nil {
-			log.Fatalf("%s", err)
+	signalChan := make(chan os.Signal)
+	signal.Notify(signalChan, syscall.SIGTERM, syscall.SIGINT)
```

Comment:
Can you please add an e2e test which creates and deletes the deployment and checks whether the rules have been cleaned ?